### PR TITLE
Adding a dropdown for buttons

### PR DESCRIPTION
**Pull request by pstjean**
Adds an optional dropdown for `prx-button` that uses content projection for the dropdown's contents.

### CODE REVIEW

File: projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css [hunk 1]
```diff
@@ -47,6 +47,49 @@ button.red:hover, button.red:focus {
...
+  display: inline-block;
+  width: 0;
+  height: 0;
+  margin-left: .255em;
+  vertical-align: .255em;
+  content: "";
+  border-top: .3em solid;
+  border-right: .3em solid transparent;
+  border-bottom: 0;
+  border-left: .3em solid transparent
```
Comment (svevang) [inline projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css]: Is this related to the icon placement?

Comment (pstjean) [reply projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css]: This creates the triangle icon (it's just CSS border properties, no SVG or anything). I swiped and modified a lot of this from Bootstrap.

Comment (pstjean) [reply projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css]: And to answer your other question (since you inexplicably can't reply to top-level comments in PRs) this is indeed a pre-req for a Publish feature! Specifically prx/publish.prx.org#579

Comment (svevang) [reply projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css]: Ahh right! `border-bottom: 0;` I assumed this was related to @rpeterman-gp work on svg/icon abstractions. Would it makes sense to use that here?

File: projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.ts [hunk 2]
```diff
@@ -11,6 +11,15 @@ import { BaseModel } from '../model/base.model';
       <ng-content></ng-content>
       <prx-spinner *ngIf="isWorking"></prx-spinner>
     </button>
+    <button *ngIf="dropdown && isVisible" type="button" class="dropdown-toggle" aria-haspopup="true"
```
Comment (svevang) [inline projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.ts]: I love the accessibility stuff here, so cool.

File: projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css (hunk 1 above)
Comment (pstjean) [reply projects/ngx-prx-styleguide/src/lib/fancy-form/button.component.css]: That would be an equally reasonable approach to this problem. Either way, it’s just a display element and there’s hidden text for accessibility.
